lib/RuleSet.js

```diff
@@ -69,277 +69,340 @@ normalized:
+		}
+		return [RuleSet.normalizeUseItem(use, ident)];
+	}
+
+	static normalizeUseItemFunction(use, data) {
+		const result = use(data);
+		if(typeof result === "string") {
+			return RuleSet.normalizeUseItem(result);
+		}
+		return result;
 	}
-	return result;
-};
 
-RuleSet.normalizeUseItemString = function normalizeUseItemString(useItemString) {
-	var idx = useItemString.indexOf("?");
-	if(idx >= 0) {
+	static normalizeUseItemString(useItemString) {
+		const idx = useItemString.indexOf("?");
+		if(idx >= 0) {
+			return {
+				loader: useItemString.substr(0, idx),
+				options: useItemString.substr(idx + 1)
+			};
+		}
 		return {
-			loader: useItemString.substr(0, idx),
-			options: useItemString.substr(idx + 1)
+			loader: useItemString
 		};
 	}
-	return {
-		loader: useItemString
-	};
-};
 
-RuleSet.normalizeUseItem = function normalizeUseItem(item, ident) {
-	if(typeof item === "function")
-		return item;
+	static normalizeUseItem(item, ident) {
+		if(typeof item === "function")
+			return item;
 
-	if(typeof item === "string") {
-		return RuleSet.normalizeUseItemString(item);
-	}
-
-	var newItem = {};
+		if(typeof item === "string") {
+			return RuleSet.normalizeUseItemString(item);
+		}
 
-	if(item.options && item.query)
-		throw new Error("Provided options and query in use");
+		let newItem = {};
 
-	if(!item.loader)
-		throw new Error("No loader specified");
+		if(item.options && item.query)
+			throw new Error("Provided options and query in use");
 
-	newItem.options = item.options || item.query;
+		if(!item.loader)
+			throw new Error("No loader specified");
 
-	if(typeof newItem.options === "object" && newItem.options) {
-		if(newItem.options.ident)
-			newItem.ident = newItem.options.ident;
-		else
-			newItem.ident = ident;
-	}
+		newItem.options = item.options || item.query;
 
-	var keys = Object.keys(item).filter(function(key) {
-		return ["options", "query"].indexOf(key) < 0;
-	});
+		if(typeof newItem.options === "object" && newItem.options) {
+			if(newItem.options.ident)
+				newItem.ident = newItem.options.ident;
+			else
+				newItem.ident = ident;
+		}
 
-	keys.forEach(function(key) {
-		newItem[key] = item[key];
-	});
+		const keys = Object.keys(item).filter(function(key) {
+			return ["options", "query"].indexOf(key) < 0;
+		});
 
-	return newItem;
-};
+		keys.forEach(function(key) {
+			newItem[key] = item[key];
+		});
 
-RuleSet.normalizeCondition = function normalizeCondition(condition) {
-	if(!condition)
-		throw new Error("Expected condition but got falsy value");
-	if(typeof condition === "string") {
-		return function(str) {
-			return str.indexOf(condition) === 0;
-		};
+		return newItem;
 	}
-	if(typeof condition === "function") {
-		return condition;
+
+	static normalizeCondition(condition) {
+		if(!condition)
+			throw new Error("Expected condition but got falsy value");
+		if(typeof condition === "string") {
+			return (str) => str.indexOf(condition) === 0;
```

Comment:
remove brackets here elsewise the linter doesn't like it: `return str => str ...`